objects/src/notes/file.rs

```diff
@@ -0,0 +1,64 @@
+use vm_core::utils::{ByteReader, ByteWriter, Deserializable, Serializable};
+use vm_processor::DeserializationError;
+
+use super::{Note, NoteDetails, NoteId, NoteInclusionProof};
+
+// NOTE FILE
+// ================================================================================================
+
+/// A serialized representation of a note.
+pub enum NoteFile {
+    /// The note has not yet been recorded on chain.
+    NoteDetails(NoteDetails),
+    /// The note has been recorded on chain.
+    NoteWithProof(Note, NoteInclusionProof),
+    /// The note's details aren't known.
+    NoteId(NoteId),
```

Comment:
minor nit: I'd move `NoteId` variant above `NoteDetails` Variant (and also order them accordingly in the serializer/deserializer).

Reply:
Done.